addons/pos_stripe/models/pos_payment_method.py

```diff
@@ -0,0 +1,110 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+import requests
+
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError, UserError
+from requests.exceptions import ConnectTimeout
+
+_logger = logging.getLogger(__name__)
+TIMEOUT = 10
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('stripe', 'Stripe')]
+
+    # Stripe
+    stripe_serial_number = fields.Char(help='[Serial number of the stripe terminal], for example: WSC513105011295', copy=False)
+
+    @api.constrains('stripe_serial_number')
+    def _check_stripe_serial_number(self):
+        if not self._get_stripe_secret_key():
+            raise ValidationError(_('Complete the Stripe onboarding.'))
+
+        for payment_method in self:
+            if not payment_method.stripe_serial_number:
+                continue
+            existing_payment_method = self.search([('id', '!=', payment_method.id),
+                                                   ('stripe_serial_number', '=', payment_method.stripe_serial_number)],
+                                                  limit=1)
+            if existing_payment_method:
+                raise ValidationError(_('Terminal %s is already used on payment method %s.')
+                                      % (payment_method.stripe_serial_number, existing_payment_method.display_name))
+
```

Comment:
nitpick: the translation function directly takes format args now, it's better to use it :)

```suggestion
                raise ValidationError(_('Terminal %s is already used on payment method %s.', payment_method.stripe_serial_number, existing_payment_method.display_name))

```